src/fparser/two/tests/fortran2003/test_write_stmt_r911.py

```diff
@@ -0,0 +1,163 @@
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''
+Test Fortran 2003 rule R911 : This file tests the support for various
+forms of WRITE statements.
+'''
+
+import pytest
+from fparser.api import get_reader
+from fparser.two.utils import FortranSyntaxError, NoMatchError
+from fparser.two.Fortran2003 import Write_Stmt, Io_Control_Spec_List, \
+    Output_Item_List, Program
+
+
+def test_write_stmt(f2003_create):
+    ''' Tests for various forms of WRITE statement. '''
+    testcls = Write_Stmt
+
+    obj = testcls('write (123)"hey"')
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == 'WRITE(123) "hey"'
+    assert repr(obj) == (
+        "Write_Stmt(Io_Control_Spec_List(',', "
+        "(Io_Control_Spec(None, Int_Literal_Constant('123', None)),)), "
+        "Char_Literal_Constant('\"hey\"', None))")
+
+    obj = testcls('WRITE (*,"(I3)") my_int')
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == 'WRITE(*, FMT = "(I3)") my_int'
+    assert repr(obj) == (
+        "Write_Stmt(Io_Control_Spec_List(',', "
+        "(Io_Control_Spec(None, Io_Unit('*')), Io_Control_Spec('FMT', "
+        "Char_Literal_Constant('\"(I3)\"', None)))), Name('my_int'))")
+
+    obj = testcls('WRITE (*,namtest)')
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == 'WRITE(*, namtest)'
+    assert repr(obj) == (
+        "Write_Stmt(Io_Control_Spec_List(',', "
+        "(Io_Control_Spec(None, Io_Unit('*')), Io_Control_Spec(None, "
+        "Name('namtest')))), None)")
+
+    # Test when format specifier contains an '=' character
```

Comment:
Shouldn't this test be part of the format specifier. This class is about the write statement and then some stuff surrrounded by brackets and then some optional stuff. It does not care what the stuff is, that is for the other classes to test.